File: MuscleHamster/Models/DailyExercise.swift
//
//  DailyExercise.swift
//  MuscleHamster
//
//  Daily check-in exercise model — one random simple exercise per day
//  Deterministic selection using stable hash (same exercise all day for a given user)
//

import Foundation

// MARK: - Daily Exercise

/// A simple bodyweight exercise for the daily check-in
struct DailyExercise: Identifiable, Equatable {
    let id: String
    let name: String
    let instruction: String
    let repCount: Int
    let icon: String  // SF Symbol name
    let encouragement: String  // Hamster-voiced encouragement

    /// Display prompt like "Do 10 Squats"
    var displayPrompt: String {
        "Do \(repCount) \(name)"
    }

    // MARK: - Exercise Pool (~35 exercises)

    static let pool: [DailyExercise] = [
        DailyExercise(id: "squats", name: "Squats", instruction: "Stand with feet shoulder-width apart, lower down like sitting in a chair, then stand back up.", repCount: 10, icon: "figure.strengthtraining.traditional", encouragement: "You crushed those squats! I'm so proud of you!"),
        DailyExercise(id: "jumping_jacks", name: "Jumping Jacks", instruction: "Jump while spreading your arms and legs, then jump back to standing.", repCount: 15, icon: "figure.jumprope", encouragement: "Woohoo! Those jumping jacks got my heart pumping too!"),
        DailyExercise(id: "push_ups", name: "Push-Ups", instruction: "Start in plank position, lower your chest to the floor, then push back up. Knees down is totally fine!", repCount: 8, icon: "figure.strengthtraining.functional", encouragement: "Amazing push-ups! You're getting stronger every day!"),
        DailyExercise(id: "lunges", name: "Lunges", instruction: "Step forward with one leg, lower your hips until both knees are bent, then step back. Alternate legs.", repCount: 10, icon: "figure.walk", encouragement: "Those lunges looked great! Your legs are going to thank you!"),
        DailyExercise(id: "plank_hold", name: "Second Plank Hold", instruction: "Hold a plank position on your forearms and toes. Keep your body in a straight line.", repCount: 20, icon: "figure.core.training", encouragement: "Wow, you held that plank like a champ! Core of steel!"),
        DailyExercise(id: "high_knees", name: "High Knees", instruction: "Jog in place, bringing your knees up toward your chest as high as comfortable.", repCount: 20, icon: "figure.run", encouragement: "Those high knees were awesome! You're a natural!"),
        DailyExercise(id: "calf_raises", name: "Calf Raises", instruction: "Stand on your toes, lifting your heels off the ground, then slowly lower back down.", repCount: 15, icon: "figure.stairs", encouragement: "Nice calf raises! Those legs are getting strong!"),
        DailyExercise(id: "arm_circles", name: "Arm Circles", instruction: "Extend your arms to the sides and make small circles. Switch direction halfway.", repCount: 20, icon: "figure.arms.open", encouragement: "Great arm circles! Feeling loose and limber!"),
        DailyExercise(id: "mountain_climbers", name: "Mountain Climbers", instruction: "In plank position, alternate driving your knees toward your chest quickly.", repCount: 12, icon: "figure.hiking", encouragement: "You climbed that mountain! Nothing can stop us!"),
        DailyExercise(id: "burpees", name: "Burpees", instruction: "Squat down, jump feet back to plank, do a push-up, jump feet forward, then jump up.", repCount: 5, icon: "figure.mixed.cardio", encouragement: "Five burpees! That's seriously impressive. You're a rockstar!"),
        DailyExercise(id: "wall_sit", name: "Second Wall Sit", instruction: "Lean against a wall with your knees at 90 degrees. Hold the position.", repCount: 20, icon: "figure.strengthtraining.traditional", encouragement: "That wall sit was tough but you did it! So strong!"),
        DailyExercise(id: "toe_touches", name: "Toe Touches", instruction: "Stand tall, then bend forward and reach for your toes. It's okay if you can't touch them!", repCount: 10, icon: "figure.flexibility", encouragement: "Great stretching! Flexibility is a superpower!"),
        DailyExercise(id: "glute_bridges", name: "Glute Bridges", instruction: "Lie on your back, knees bent, feet flat. Lift your hips toward the ceiling, squeeze, then lower.", repCount: 12, icon: "figure.pilates", encouragement: "Those bridges were perfect! Your glutes are thanking you!"),
        DailyExercise(id: "bicycle_crunches", name: "Bicycle Crunches", instruction: "Lie on your back, hands behind your head. Alternate touching your elbow to the opposite knee.", repCount: 12, icon: "figure.core.training", encouragement: "Awesome crunches! Your core is getting so strong!"),
        DailyExercise(id: "standing_side_bends", name: "Side Bends", instruction: "Stand with feet hip-width apart, reach one arm overhead and lean to the opposite side. Alternate.", repCount: 10, icon: "figure.flexibility", encouragement: "Nice side bends! That's great for your obliques!"),
        DailyExercise(id: "step_ups", name: "Step-Ups", instruction: "Find a sturdy step or low chair. Step up with one foot, bring the other up, then step down. Alternate.", repCount: 10, icon: "figure.stairs", encouragement: "Those step-ups were great! One step at a time!"),
        DailyExercise(id: "tricep_dips", name: "Tricep Dips", instruction: "Use a sturdy chair edge. Lower your body by bending your elbows, then push back up.", repCount: 8, icon: "figure.strengthtraining.functional", encouragement: "Awesome dips! Those arms are getting toned!"),
        DailyExercise(id: "donkey_kicks", name: "Donkey Kicks", instruction: "On all fours, kick one leg back and up toward the ceiling. Keep your core tight. Alternate legs.", repCount: 10, icon: "figure.pilates", encouragement: "Great donkey kicks! Your hamster is kicking along with you!"),
        DailyExercise(id: "superman_holds", name: "Second Superman Hold", instruction: "Lie face down, extend arms forward. Lift arms, chest, and legs off the ground simultaneously.", repCount: 15, icon: "figure.core.training", encouragement: "You looked like a superhero! Back muscles activated!"),
        DailyExercise(id: "jumping_squats", name: "Jump Squats", instruction: "Do a squat, then explode upward into a jump. Land softly and repeat.", repCount: 8, icon: "figure.jumprope", encouragement: "Those jump squats were explosive! You're getting so powerful!"),
        DailyExercise(id: "hip_circles", name: "Hip Circles", instruction: "Stand on one leg and make circles with the other hip. Switch legs halfway.", repCount: 10, icon: "figure.cooldown", encouragement: "Nice hip circles! Keeping those joints happy and healthy!"),
        DailyExercise(id: "leg_raises", name: "Leg Raises", instruction: "Lie on your back, legs straight. Lift both legs up toward the ceiling, then slowly lower.", repCount: 10, icon: "figure.core.training", encouragement: "Strong leg raises! Your core is really working hard!"),
        DailyExercise(id: "inchworms", name: "Inchworms", instruction: "Stand tall, bend forward, walk hands out to plank, then walk hands back and stand up.", repCount: 5, icon: "figure.flexibility", encouragement: "Inchworms are so fun! Great full-body stretch!"),
        DailyExercise(id: "side_lunges", name: "Side Lunges", instruction: "Step wide to one side, bending that knee while keeping the other leg straight. Alternate.", repCount: 8, icon: "figure.walk", encouragement: "Those side lunges hit different! Great inner thigh work!"),
        DailyExercise(id: "flutter_kicks", name: "Flutter Kicks", instruction: "Lie on your back, lift both legs slightly. Alternate kicking up and down in small movements.", repCount: 15, icon: "figure.core.training", encouragement: "Flutter kicks done! Your abs are on fire (in a good way)!"),
        DailyExercise(id: "shoulder_taps", name: "Shoulder Taps", instruction: "In plank position, tap your left shoulder with your right hand, then switch. Stay stable!", repCount: 10, icon: "figure.strengthtraining.functional", encouragement: "Great shoulder taps! Balance and strength combined!"),
        DailyExercise(id: "reverse_lunges", name: "Reverse Lunges", instruction: "Step one foot backward, lower into a lunge, then return to standing. Alternate legs.", repCount: 10, icon: "figure.walk", encouragement: "Reverse lunges are so good for your legs! Well done!"),
        DailyExercise(id: "seated_twist", name: "Seated Twists", instruction: "Sit on the floor, lean back slightly, and twist your torso side to side. Keep your core engaged.", repCount: 12, icon: "figure.core.training", encouragement: "Those twists felt great, right? Obliques engaged!"),
        DailyExercise(id: "standing_march", name: "Standing Marches", instruction: "March in place, lifting your knees high and swinging your arms. Nice and steady.", repCount: 20, icon: "figure.walk", encouragement: "Great marching! Every step counts toward a healthier you!"),
        DailyExercise(id: "skater_hops", name: "Skater Hops", instruction: "Hop side to side on one foot, swinging your arms like a speed skater.", repCount: 10, icon: "figure.mixed.cardio", encouragement: "Those skater hops were awesome! Great for balance!"),
        DailyExercise(id: "dead_bugs", name: "Dead Bugs", instruction: "Lie on your back, arms up, knees bent at 90 degrees. Extend opposite arm and leg, then switch.", repCount: 10, icon: "figure.pilates", encouragement: "Dead bugs are sneaky hard! Your core did amazing!"),
        DailyExercise(id: "bird_dogs", name: "Bird Dogs", instruction: "On all fours, extend your right arm and left leg simultaneously. Hold, then switch sides.", repCount: 8, icon: "figure.core.training", encouragement: "Bird dogs are so good for balance and core! Nailed it!"),
        DailyExercise(id: "lateral_raises_bodyweight", name: "Arm Lateral Raises", instruction: "Stand with arms at your sides. Raise them out to the sides until shoulder height, then lower slowly.", repCount: 12, icon: "figure.arms.open", encouragement: "Great lateral raises! Those shoulders are getting strong!"),
        DailyExercise(id: "knee_push_ups", name: "Knee Push-Ups", instruction: "Like regular push-ups but with knees on the ground. Lower your chest, then push back up.", repCount: 10, icon: "figure.strengthtraining.functional", encouragement: "Perfect form! Knee push-ups are just as awesome!"),
        DailyExercise(id: "standing_crunches", name: "Standing Crunches", instruction: "Stand tall, bring one knee up while crunching your elbow toward it. Alternate sides.", repCount: 12, icon: "figure.core.training", encouragement: "Standing crunches are underrated! Great job!"),
    ]

    // MARK: - Deterministic Daily Selection

    /// Get today's exercise for a specific user (deterministic — same all day)
    static func todaysExercise(for userId: String) -> DailyExercise {
        let calendar = Calendar.current
        let now = Date()
        let dayOfYear = calendar.ordinality(of: .day, in: .year, for: now) ?? 1
        let year = calendar.component(.year, from: now)

        // Use djb2 hash on userId bytes + day-of-year + year for stable, deterministic selection
        let seed = djb2Hash(userId: userId, dayOfYear: dayOfYear, year: year)

        // Map to pool index
        let index = Int(seed % UInt64(pool.count))
        return pool[index]
    }

    /// djb2 hash — stable across runs (unlike Swift's hashValue which is randomized)
    private static func djb2Hash(userId: String, dayOfYear: Int, year: Int) -> UInt64 {
        let input = "\(userId)_\(dayOfYear)_\(year)"
        var hash: UInt64 = 5381
        for byte in input.utf8 {
            hash = ((hash &<< 5) &+ hash) &+ UInt64(byte)  // hash * 33 + byte
        }
        return hash
    }
}

// MARK: - Daily Exercise Check-In Record

/// Records a completed daily exercise check-in
struct DailyExerciseCheckIn: Codable, Identifiable, Equatable {
    let id: String
    let exerciseId: String
    let exerciseName: String
    let completedAt: Date
    let pointsEarned: Int

    var displayDate: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .short
        return formatter.string(from: completedAt)
    }
}

File: MuscleHamsterExpo/src/models/DailyExercise.js
// Daily Exercise Model
// "The Daily 36" — a balanced mix of strength, cardio, and mobility exercises
// One random exercise per day, deterministic per user

// djb2 hash — stable across runs (unlike JS's Math.random)
const djb2Hash = (str) => {
  let hash = 5381;
  for (let i = 0; i < str.length; i++) {
    hash = ((hash << 5) + hash + str.charCodeAt(i)) >>> 0;
  }
  return hash;
};

// Get day-of-year (1-based)
const dayOfYear = (date = new Date()) => {
  const start = new Date(date.getFullYear(), 0, 0);
  const diff = date - start;
  return Math.floor(diff / (1000 * 60 * 60 * 24));
};

// Exercise pool — "The Daily 36"
// A balanced mix of strength, cardio, mobility, and breathing exercises
export const dailyExercisePool = [
  {
    id: 'classicPushUps',
    name: 'Classic Push-Ups',
    repCount: 20,
    repType: 'reps',
    instruction: 'Start in a plank position with hands shoulder-width apart. Lower your chest to the ground, then push back up. Keep your body in a straight line throughout.',
    icon: 'fitness',
    encouragement: "Push it! You're building real strength!",
  },
  {
    id: 'airSquats',
    name: 'Air Squats',
    repCount: 30,
    repType: 'reps',
    instruction: 'Stand with feet shoulder-width apart. Lower your hips back and down like sitting in a chair, then stand back up. Keep your chest up and knees over toes.',
    icon: 'body',
    encouragement: "Squat it out! Your legs are getting stronger!",
  },
  {
    id: 'forearmPlank',
    name: 'Forearm Plank',
    repCount: 60,
    repType: 'seconds',
    instruction: 'Rest on your forearms and toes, keeping your body in a straight line from head to heels. Engage your core and hold steady.',
    icon: 'remove',
    encouragement: "Hold strong! Your core is working hard!",
  },
  {
    id: 'bicycleCrunches',
    name: 'Bicycle Crunches',
    repCount: 40,
    repType: 'reps',
    instruction: 'Lie on your back with hands behind your head. Bring opposite elbow to opposite knee while extending the other leg. Alternate sides in a pedaling motion.',
    icon: 'bicycle',
    encouragement: "Pedal those abs! You're crushing it!",
  },
  {
    id: 'wallSit',
    name: 'Wall Sit',
    repCount: 60,
    repType: 'seconds',
    instruction: 'Lean your back against a wall and slide down until your thighs are parallel to the ground. Hold this seated position against the wall.',
    icon: 'square',
    encouragement: "Hold it! Feel that burn—you've got this!",
  },
  {
    id: 'catCowFlow',
    name: 'Cat-Cow Flow',
    repCount: 10,
    repType: 'reps',
    instruction: 'On hands and knees, alternate between arching your back up like a cat and letting your belly sink while lifting your head. Move smoothly between positions.',
    icon: 'paw',
    encouragement: "Flow with it! Your spine is so happy!",
  },
  {
    id: 'jumpingJacks',
    name: 'Jumping Jacks',
    repCount: 50,
    repType: 'reps',
    instruction: 'Start standing with arms at your sides. Jump while spreading your legs and raising your arms overhead, then jump back to start.',
    icon: 'accessibility',
    encouragement: "Jump jump! Getting that heart pumping!",
  },
  {
    id: 'childsPose',
    name: "Child's Pose",
    repCount: 60,
    repType: 'seconds',
    instruction: 'Kneel on the floor, sit back on your heels, and stretch your arms forward on the ground. Rest your forehead down and breathe deeply.',
    icon: 'bed',
    encouragement: "Rest and restore. This is self-care at its finest!",
  },
  {
    id: 'highKnees',
    name: 'High Knees',
    repCount: 60,
    repType: 'seconds',
    instruction: 'Run in place while driving your knees up toward your chest as high as you can. Pump your arms and keep a quick pace.',
    icon: 'walk',
    encouragement: "Knees up! You're a cardio machine!",
  },
  {
    id: 'standingQuadStretch',
    name: 'Standing Quad Stretch',
    repCount: 30,
    repType: 'seconds',
    instruction: 'Stand on one leg and pull your other heel toward your glute. Hold for 30 seconds, then switch legs. Use a wall for balance if needed.',
    icon: 'body',
    encouragement: "Stretch it out! 30 seconds per leg—you've earned this!",
  },

  // ===== CARDIO & MOVEMENT =====
  {
    id: 'buttKicks',
    name: 'Butt Kicks',
    repCount: 60,
    repType: 'seconds',
    instruction: 'Jog in place while kicking your heels up toward your glutes with each step. Keep a quick, steady rhythm.',
    icon: 'footsteps',
    encouragement: "Kick it up! Your legs are on fire!",
  },
  {
    id: 'skaterJumps',
    name: 'Skater Jumps',
    repCount: 30,
    repType: 'reps',
    instruction: 'Leap sideways from one foot to the other, landing softly and swinging your arms for momentum. Like a speed skater!',
    icon: 'swap-horizontal',
    encouragement: "Glide like a pro! Side to side power!",
  },
  {
    id: 'sealJacks',
    name: 'Seal Jacks',
    repCount: 40,
    repType: 'reps',
    instruction: 'Like jumping jacks, but clap your arms in front of you instead of overhead. Legs jump out and in as usual.',
    icon: 'accessibility',
    encouragement: "Clap clap! Seal of approval!",
  },
  {
    id: 'sprintingInPlace',
    name: 'Sprinting in Place',
    repCount: 45,
    repType: 'seconds',
    instruction: 'Run in place as fast as you can! Pump your arms, drive your knees high, and give it maximum effort.',
    icon: 'flash',
    encouragement: "Go go go! Maximum effort sprint!",
  },

  // ===== STRENGTH & STABILITY =====
  {
    id: 'goodMornings',
    name: 'Good Mornings',
    repCount: 20,
    repType: 'reps',
    instruction: 'Stand with hands behind your head. Keeping your back straight, hinge at the hips until your torso is nearly parallel to the floor, then return to standing.',
    icon: 'sunny',
    encouragement: "Good morning to your hamstrings!",
  },
  {
    id: 'standingSideCrunches',
    name: 'Standing Side Crunches',
    repCount: 30,
    repType: 'reps',
    instruction: 'Stand tall, lift one knee to the side while bringing your elbow down to meet it. Alternate sides with each rep.',
    icon: 'body',
    encouragement: "Crunch those obliques! Side to side!",
  },
  {
    id: 'tricepFloorDips',
    name: 'Tricep Floor Dips',
    repCount: 25,
    repType: 'reps',
    instruction: 'Sit on the floor with hands behind you, fingers pointing forward. Lift your hips slightly and perform small pulsing dips.',
    icon: 'fitness',
    encouragement: "Feel those triceps working!",
  },
  {
    id: 'plankShoulderTaps',
    name: 'Plank Shoulder Taps',
    repCount: 40,
    repType: 'reps',
    instruction: 'In a high plank position, slowly tap your left shoulder with your right hand, then alternate. Keep your hips stable—no rocking!',
    icon: 'hand-right',
    encouragement: "Tap tap! Core stability champion!",
  },
  {
    id: 'wallAngels',
    name: 'Wall Angels',
    repCount: 15,
    repType: 'reps',
    instruction: 'Stand with your back against a wall, elbows and wrists touching the wall. Slowly slide your arms up and down like making a snow angel.',
    icon: 'body',
    encouragement: "Angel wings! Your posture thanks you!",
  },
  {
    id: 'diverPushUps',
    name: 'Diver Push-Ups',
    repCount: 8,
    repType: 'reps',
    instruction: 'Start in Downward Dog. Scoop your chest low to the floor in a wave motion, then arch up into Upward Dog. Reverse the motion to return.',
    icon: 'fitness',
    encouragement: "Dive deep! Full body flow!",
  },
  {
    id: 'proneYRaise',
    name: 'Prone Y-Raise',
    repCount: 20,
    repType: 'reps',
    instruction: 'Lie face down with arms extended in a Y shape. Lift both arms off the floor, squeezing your upper back, then lower slowly.',
    icon: 'arrow-up',
    encouragement: "Y so strong? Because you are!",
  },
  {
    id: 'handMarches',
    name: 'Hand Marches',
    repCount: 60,
    repType: 'seconds',
    instruction: 'In a high plank position, lift one hand slightly off the floor, then the other, like marching in place with your hands.',
    icon: 'hand-left',
    encouragement: "March those hands! Plank power!",
  },
  {
    id: 'shoulderProtraction',
    name: 'Shoulder Protraction',
    repCount: 15,
    repType: 'reps',
    instruction: 'In a plank position, push your spine toward the ceiling rounding your upper back, then let it sink back down. Control the movement.',
    icon: 'body',
    encouragement: "Push and sink! Scapula strength!",
  },
  {
    id: 'cobraPulses',
    name: 'Cobra Pulses',
    repCount: 20,
    repType: 'reps',
    instruction: 'Lie face down with hands by your shoulders. Using your back muscles (not your hands), lift your chest slightly off the floor in small pulses.',
    icon: 'arrow-up',
    encouragement: "Rise up! Strong back, proud chest!",
  },

  // ===== MOBILITY & BREATHING =====
  {
    id: 'boxBreathing',
    name: 'Box Breathing',
    repCount: 4,
    repType: 'reps',
    instruction: 'Breathe in for 4 seconds, hold for 4 seconds, breathe out for 4 seconds, hold for 4 seconds. That\'s one cycle. Complete 4 cycles.',
    icon: 'square',
    encouragement: "Breathe... calm... focus... peace...",
  },
  {
    id: 'balanceTest',
    name: 'Balance Test',
    repCount: 30,
    repType: 'seconds',
    instruction: 'Stand on one leg for 30 seconds, then switch to the other leg. Keep your core engaged and find a focal point.',
    icon: 'body',
    encouragement: "Steady now! Balance mastery!",
  },
  {
    id: 'wristCircles',
    name: 'Wrist Circles',
    repCount: 30,
    repType: 'seconds',
    instruction: 'Extend your arms and rotate your wrists in circles. Do 30 seconds in each direction to loosen up those joints.',
    icon: 'sync',
    encouragement: "Round and round! Happy wrists!",
  },
  {
    id: 'ankleCircles',
    name: 'Ankle Circles',
    repCount: 30,
    repType: 'seconds',
    instruction: 'Lift one foot off the ground and rotate your ankle in circles. 30 seconds per foot, both directions.',
    icon: 'sync',
    encouragement: "Circle those ankles! Mobile feet!",
  },
  {
    id: 'threadTheNeedle',
    name: 'Thread the Needle',
    repCount: 30,
    repType: 'seconds',
    instruction: 'On hands and knees, reach one arm under your body and rotate your torso, resting your shoulder on the ground. 30 seconds per side.',
    icon: 'refresh',
    encouragement: "Thread it through! Spine feels amazing!",
  },
  {
    id: 'deepSquatHold',
    name: 'Deep Squat Hold',
    repCount: 60,
    repType: 'seconds',
    instruction: 'Sink into the bottom of a squat and hold. Keep your heels down, chest up, and breathe. This is primal mobility!',
    icon: 'body',
    encouragement: "Sink low and hold! Primal strength!",
  },
  {
    id: 'calfStretchWall',
    name: 'Calf Stretch (Wall)',
    repCount: 30,
    repType: 'seconds',
    instruction: 'Face a wall, place one foot back with heel down, and lean forward until you feel a stretch in your calf. 30 seconds per leg.',
    icon: 'body',
    encouragement: "Stretch those calves! They work hard for you!",
  },
  {
    id: 'doorwayLatStretch',
    name: 'Doorway Lat Stretch',
    repCount: 30,
    repType: 'seconds',
    instruction: 'Reach one arm high on a doorframe and lean your body away until you feel a stretch along your side. 30 seconds per side.',
    icon: 'home',
    encouragement: "Open up those lats! Feel the length!",
  },

  // ===== POWER MOVES =====
  {
    id: 'airPunchesSpeed',
    name: 'Air Punches (Speed)',
    repCount: 60,
    repType: 'seconds',
    instruction: 'Throw straight punches into the air as fast as you can! Alternate hands rapidly. Maximum speed, stay light on your feet.',
    icon: 'flash',
    encouragement: "Punch punch punch! Speed demon!",
  },
];

// The Final Stretch - shown after every check-in, not a random exercise
export const FINAL_STRETCH_PROMPT = {
  title: 'The Final Stretch',
  instruction: 'Take a massive breath in, reach as high as you can, and exhale everything out.',
  icon: 'sunny',
};

// Get today's exercise for a user (deterministic)
export const getTodaysExercise = (userId) => {
  const now = new Date();
  const doy = dayOfYear(now);
  const year = now.getFullYear();
  const seed = `${userId}-${doy}-${year}`;
  const hash = djb2Hash(seed);
  const index = hash % dailyExercisePool.length;
  return dailyExercisePool[index];
};

// Computed display prompt based on rep type
export const getExerciseDisplayPrompt = (exercise) => {
  const { repCount, repType, name } = exercise;

  switch (repType) {
    case 'seconds':
      return `Hold for ${repCount} Seconds`;
    case 'minutes':
      return `Do for ${repCount} Minute${repCount > 1 ? 's' : ''}`;
    case 'breath':
      return 'Take a Deep Breath';
    case 'reps':
    default:
      return `Do ${repCount} ${name}`;
  }
};

// Create a daily exercise check-in record
export const createDailyExerciseCheckIn = ({
  id,
  exerciseId,
  exerciseName,
  completedAt,
  pointsEarned,
}) => ({
  id,
  exerciseId,
  exerciseName,
  completedAt,
  pointsEarned,
});
